src/libraries/System.Text.RegularExpressions/gen/UpgradeToRegexGeneratorCodeFixer.cs

```diff
@@ -103,16 +103,21 @@ private static async Task<Document> ConvertToSourceGenerator(Document document,
             }
 
             // Get the parent type declaration so that we can inspect its methods as well as check if we need to add the partial keyword.
-            TypeDeclarationSyntax? typeDeclaration = nodeToFix.Ancestors().OfType<TypeDeclarationSyntax>().FirstOrDefault();
+            SyntaxNode? typeDeclarationOrCompilationUnit = nodeToFix.Ancestors().OfType<TypeDeclarationSyntax>().FirstOrDefault();
 
-            if (typeDeclaration is null)
+            if (typeDeclarationOrCompilationUnit is null)
             {
-                return document;
+                typeDeclarationOrCompilationUnit = await nodeToFix.SyntaxTree.GetRootAsync(cancellationToken).ConfigureAwait(false);
             }
 
             // Calculate what name should be used for the generated static partial method
             string methodName = DefaultRegexMethodName;
-            ITypeSymbol? typeSymbol = semanticModel.GetDeclaredSymbol(typeDeclaration, cancellationToken) as ITypeSymbol;
+            INamedTypeSymbol? typeSymbol = typeDeclarationOrCompilationUnit switch
+            {
+                TypeDeclarationSyntax typeDeclaration => semanticModel.GetDeclaredSymbol(typeDeclaration, cancellationToken),
+                CompilationUnitSyntax compilationUnit => semanticModel.GetDeclaredSymbol(compilationUnit, cancellationToken)?.ContainingType,
```

Comment:
For top-level case, do we still need to do all of the logic from lines 131-152? Seems to me that for that case we should be doing something different, right?

Reply:
I see, makes sense since there are no ancestors of TypeDeclarationSyntax. I think it's fine to leave as is for now.